sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/sas/TableAccountSasResourceType.java

```diff
@@ -0,0 +1,157 @@
+            char c = resourceTypesString.charAt(i);
+
+            switch (c) {
+                case 's':
+                    resourceType.service = true;
+                    break;
+                case 'c':
+                    resourceType.container = true;
+                    break;
+                case 'o':
+                    resourceType.object = true;
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                        String.format(Locale.ROOT, StorageConstants.ENUM_COULD_NOT_BE_PARSED_INVALID_VALUE,
+                            "Resource Types", resourceTypesString, c));
+            }
+        }
+
+        return resourceType;
+    }
+
+    /**
+     * Get the access status for service level APIs.
+     *
+     * @return The access status for service level APIs.
+     */
+    public boolean isService() {
+        return service;
+    }
+
+    /**
+     * Sets the access status for service level APIs.
+     *
+     * @param service The access status to set.
+     *
+     * @return The updated {@link TableAccountSasResourceType} object.
+     */
+    public TableAccountSasResourceType setService(boolean service) {
+        this.service = service;
+
+        return this;
+    }
+
+    /**
+     * Gets the access status for container level APIs, this grants access to Blob Containers, Tables, Queues, and
+     * File Shares.
+     *
+     * @return The access status for container level APIs, this grants access to Blob Containers, Tables, Queues, and
+     * File Shares.
+     */
+    public boolean isContainer() {
+        return container;
+    }
+
+    /**
+     * Sets the access status for container level APIs, this grants access to Blob Containers, Tables, Queues, and File
+     * Shares.
+     *
+     * @param container The access status to set.
+     *
+     * @return The updated {@link TableAccountSasResourceType} object.
+     */
+    public TableAccountSasResourceType setContainer(boolean container) {
+        this.container = container;
+
+        return this;
+    }
+
+    /**
+     * Get the access status for object level APIs, this grants access to Blobs, Table Entities, Queue Messages, Files.
+     *
+     * @return The access status for object level APIs, this grants access to Blobs, Table Entities, Queue Messages,
+     * Files.
+     */
+    public boolean isObject() {
+        return object;
+    }
+
+    /**
+     * Sets the access status for object level APIs, this grants access to Blobs, Table Entities, Queue Messages,
+     * Files.
+     *
+     * @param object The access status to set.
+     *
+     * @return The updated {@link TableAccountSasResourceType} object.
+     */
+    public TableAccountSasResourceType setObject(boolean object) {
+        this.object = object;
+
+        return this;
+    }
+
+    /**
+     * Converts the given resource types to a {@link String}. Using this method will guarantee the resource types are in
```

Comment:
We should document that if all resource types are set to `false`, an empty string is returned from this method.